src/dawn/Optimizer/StatementMapper.cpp

```diff
@@ -120,37 +120,42 @@ void StatementMapper::visit(const std::shared_ptr<iir::IfStmt>& stmt) {
 void StatementMapper::visit(const std::shared_ptr<iir::VarDeclStmt>& stmt) {
   DAWN_ASSERT(initializedWithBlockStmt_);
 
-  // This is the first time we encounter this variable. We have to make sure the name is not
-  // already used in another scope!
-  int AccessID = instantiation_->nextUID();
+  int accessID = -1;
+  if(metadata_.hasStmtToAccessID(stmt)) {
```

Comment:
is this change needed because after duplicating a do-method, we might have two local variables with the same name?